SCF-Session-1/ProductPromotion.java

```diff
@@ -0,0 +1,73 @@
+package shoppingCart;
+
+import java.util.Calendar;
+
+public class ProductPromotion implements Promotion
+{
+	int minQuantity;
+	double fixedDiscount, minimumPrice;
+	 
+	ProductPromotion(int minQuantity) 
+	{
+		this.minQuantity = minQuantity;					// setting up minimum price and no of products limit to avail discount
+		setFixedDiscount();										// also declaring discount percent
+	}
+	
+	public void setFixedDiscount()
+	{
+		fixedDiscount = 0.05;
+	}
+	
+	public double getFixedDiscount()
+	{
+	    return fixedDiscount;
+	}
+	
+	public int getMinQuantity()
+	{
+	    return minQuantity;
+	}
+	
+	public void setMinimumPrice(double minPrice)
+	{
+		this.minimumPrice = minPrice;
+	}
+	
+	public void setFixedDiscount(double fixedDiscount)
+	{
+		this.fixedDiscount = fixedDiscount;
+	}
+	
+	public double getMinimumPrice()
+	{
+		return minimumPrice;
+	}
+	
+	public boolean isPromotionApplicable(double totalQuantity, String userCode)
+	{
```

Comment:
This method `isPromotionApplicable()` is a bit confusing. Please elaborate a bit about the functionality.